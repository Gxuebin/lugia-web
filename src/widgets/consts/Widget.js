/**
 *
 * create by ligx
 *
 * @flow
 */
export const Input = 'sv.widget.Input';
export const Align = 'sv.widget.Align';
export const Trigger = 'sv.widget.Trigger';
export const InputTag = 'sv.widget.InputTag';
export const Theme = 'sv.widget.Theme';
export const Menu = 'sv.widget.Menu';
export const DropMenu = 'sv.widget.DropMenu';
export const Tree = 'sv.widget.Tree';
export const ThrottleScroller = 'sv.widget.ThrottleScroller';
