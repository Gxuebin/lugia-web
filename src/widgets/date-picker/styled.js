import styled from 'styled-components';
import colorsFunc from '../css/stateColor';
import { px2emcss } from '../css/units';
import { valueInRange } from '../common/Math';
const em = px2emcss(1.2);
const distance = {
  iconLeft: 10,
};
const DateWrapperPadding = {
  top: 30,
  left: 20,
  bottom: 44,
};
const { hoverColor, normalColor } = colorsFunc();

export const Icons = styled.span`
  position: absolute;
  left: ${em(distance.iconLeft)};
  top: 50%;
  transform: translateY(-50%);
`;
export const DateWrapper = styled.div`
  display: inline-block;
  width: ${props => props.width}px;
  padding: ${props => getDateWrrap(props).paddingStyle};
  border: 1px solid #ddd;
`;

export const DateHeader = styled.div`
  font-size: 12px;
`;
export const RangeInput = styled.div`
  display: inline-block;
  border: 1px solid #e8e8e8;
  border-radius: 3px;

  & input {
    width: 200px;
    border: none;
    text-align: center;
  }
  & input:focus {
    border: none;
    box-shadow: none;
  }
`;
export const RangeSpan = styled.span`
  font-size: 12px;
`;
export const HeaderTop = styled.div`
  text-align: center;
  margin-bottom: 24px;
`;
export const HeaderTopText = styled.span`
  color: #333;
  font-size: 14px;
  font-weight: 500;
  cursor: pointer;
  &:hover {
    color: ${hoverColor};
  }
`;
export const HeaderTopArrow = HeaderTopText.extend`    
    float:${props => props.position};
    margin-${props => props.position}:${props => props.margin}px;
    vertical-align:middle;
`;
export const HeaderWeekBox = styled.ul`
  padding: 0;
  margin: 0;
  margin-bottom: 12px;
`;
export const HeaderWeek = styled.li`
  display: inline-block;
  width: ${props => em((props.width - DateWrapperPadding.left * 2 - 2) / 7)};
  height: ${props => em((props.width - DateWrapperPadding.left * 2 - 2) / 7)};
  text-align: center;
  line-height: ${props => em((props.width - DateWrapperPadding.left * 2 - 2) / 7)};
  cursor: pointer;
`;
export const DatePanel = styled.div`
  font-size: 12px;
`;
const dateSize = {
  DateChildWidth: 26,
};
export const DateChild = styled.span`
  display: inline-block;
  width: ${props => em((props.width - DateWrapperPadding.left * 2 - 2) / 7)};
  text-align: center;
  vertical-align: middle;
  margin: ${em(3)} ${0};
  ${props => getDateChildStyle(props).chooseStyle};
  ${props => getDateChildStyle(props).chooseWeeks};
  ${props => getDateChildStyle(props).rangeStyle};
  ${props => getDateChildStyle(props).todayStyle};
`;
export const DateChildInner = styled.i`
  font-style: normal;
  display: inline-block;
  width: ${em(dateSize.DateChildWidth)};
  height: ${em(dateSize.DateChildWidth)};
  text-align: center;
  line-height: ${em(dateSize.DateChildWidth)};
  vertical-align: text-top;
  cursor: pointer;

  &:hover {
    background: ${hoverColor};
  }

  color: ${props => (props.outMonth ? '#ccc' : '#666')};
`;
export const OtherChild = styled.span`
  display: inline-block;
  width: ${props => 100 / props.column}%;
  line-height: ${em(40)};
  font-size: 14px;
  text-align: center;
  white-space: nowrap;

  &:hover {
    color: ${hoverColor};
  }

  cursor: pointer;
`;
export const OtherChildText = styled.i`
  padding: 5px 10px;
  font-style: normal;
  border-radius: 3px;
  ${props => (props.isChose ? `background:${normalColor};color:#fff;` : '')};
`;
export const RangeWrap = styled.div`
  display: inline-block;
`;
export const RangeInnerTop = styled.span`
  display: block;
`;
export const RangeInputWrap = styled.span`
  display: inline-block;
  border: 1px solid #ddd;
  border-radius: 3px;
`;
export const RangeInputInner = styled.span`
  & input {
    border: none;
    text-align: center;
  }
  & input:focus {
    border: none;
    box-shadow: none;
  }
`;
const getDateWrrap = props => {
  const { top, left, bottom } = DateWrapperPadding;
  const paddingStyle = ` ${em(top)}  ${em(left)} ${em(bottom)}`;
  return {
    paddingStyle,
  };
};
const getDateChildStyle = props => {
  const {
    choseDayIndex,
    isChooseWeek,
    startInWeeks,
    endInWeeks,
    isHoverWeek,
    weekHoverStart,
    weekHoverEnd,
    rangeChose,
    rangeIndex,
    index,
    todayIndex,
    noToday,
    rangeStartIndex,
    rangeEndIndex,
    panelFistEndIndex,
    panelSecondStartIndex,
  } = props;

  const arrChoseDayIndex = Array.isArray(choseDayIndex) ? choseDayIndex : [choseDayIndex];

  const chooseStyle = arrChoseDayIndex.reduce((p, n) => {
    return `${p}
    &:nth-child(${n})>i{
      background:${normalColor};
      color:#fff;
      border-radius:50%;
    }`;
  }, '');
  const todayInd = noToday ? '' : todayIndex;
  let todayStyle = `
      &:nth-child(${todayInd})>i{
        border:1px solid ${normalColor};
        border-radius:50%;
      }
  `;
  let chooseWeeks;
  let chooseWeekRadius;
  if (isChooseWeek || isHoverWeek) {
    const backG = isChooseWeek ? `${normalColor}` : '#8f83ff';
    const start = isChooseWeek ? startInWeeks + 1 : weekHoverStart + 1;
    const end = isChooseWeek ? endInWeeks : weekHoverEnd;
    const todayIn = valueInRange(todayIndex, [start, end]);
    if (todayIn) {
      todayStyle = `
      &:nth-child(${todayInd})>i{
        border:1px solid transparent;
        border-radius:50%;       
      }
  `;
    }
    chooseWeeks = `
    background:${backG};
    
      &>i{
        color:#fff;
        border-radius:50%;
      }  

      &:nth-child(${start}){
        border-top-left-radius:20px;
        border-bottom-left-radius:20px;
      }
  
      &:nth-child(${end}){
        border-top-right-radius:20px;
        border-bottom-right-radius:20px;
      } 
    `;
  }
  let rangeStyle;
  if (rangeChose) {
    let isEnd = false;
    const startOrEnd = index % 7;
    let borderIndex;
    if (startOrEnd === 0) {
      isEnd = true;
      borderIndex = index;
    }
    if (startOrEnd === 1) {
      isEnd = false;
      borderIndex = index;
    }

    const direction = rangeIndex === 0 ? 'left' : 'right';
    const startDir = isEnd ? 'right' : 'left';
    const borderStyle = (dire: string) => {
      return `
        border-top-${dire}-radius:20px;
        border-bottom-${dire}-radius:20px;
      `;
    };
    rangeStyle = `    
      background:#684fff1f;
      &:nth-child(${arrChoseDayIndex}){
        border-top-${direction}-radius:20px; 
        border-bottom-${direction}-radius:20px;
      }    
      &:nth-child(${borderIndex}){
        border-top-${startDir}-radius:20px;
        border-bottom-${startDir}-radius:20px;
      }  
     
      &:nth-child(${rangeStartIndex}){       
        ${borderStyle('left')}
      } 
      &:nth-child(${rangeEndIndex}){        
        ${borderStyle('right')}
      }
      &:nth-child(${panelFistEndIndex}){        
        ${borderStyle('right')}
      }
      &:nth-child(${panelSecondStartIndex}){        
        ${borderStyle('left')}
      }
    `;
  }
  return {
    chooseStyle,
    chooseWeeks,
    chooseWeekRadius,
    rangeStyle,
    todayStyle,
  };
};
