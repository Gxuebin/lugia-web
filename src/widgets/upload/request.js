/*
 *create by LYQ
 *
 *2018-12-05
 *
 *@flow
 *
 */

export function getRequestXHR(): Object {
  return window.XMLHttpRequest
    ? new XMLHttpRequest()
    : new window.ActiveXobject('Microsoft.XMLHTTP');
}

export function getParamsData(data: Object, file: Object): Object {
  const newData: Object = new FormData();

  for (const i in data) {
    newData.append(i, data[i]);
  }
  newData.append('file', file);
  return newData;
}

export function getStringFromObject(data: ?Object): string {
  if (!data) return '';
  let resultString = '';
  for (const i in data) {
    resultString += i + '=' + data[i] + '&';
  }
  resultString = resultString.slice(0, -1);
  return resultString;
}

function addEventListener(
  target: Object,
  event: string,
  func: Function,
  useCapture?: boolean = false
) {
  target.addEventListener(event, func, useCapture);
}

export function request(dataObject: Object): Object {
  const { url } = dataObject;
  if (!url) {
    return;
  }
  const xhr = getRequestXHR();
  const { withCredentials = false } = dataObject;
  xhr.withCredentials = withCredentials;

  const { method = 'get', asynch = true, data, file } = dataObject;
  const params = getParamsData(data, file);

  const { onProgress, onComplete } = dataObject;
  xhr.upload.onprogress = onProgress;
  if (onProgress) addEventListener(xhr.upload, 'progress', onProgress, false);
  if (onComplete) addEventListener(xhr, 'load', onComplete, false);

  if (method.toLocaleLowerCase() === 'get') {
    if (data) {
      xhr.open('get', url + '?' + getStringFromObject(params), asynch);
    } else {
      xhr.open('get', url, asynch);
    }
    xhr.send();
  }
  if (method.toLocaleLowerCase() === 'post') {
    xhr.open('post', url, asynch);
    const { headers } = dataObject;
    if (headers) {
      Object.keys(headers).forEach(k => {
        xhr.setRequestHeader(k, headers[k]);
      });
    } else {
      // xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
    }
    xhr.send(params);
  }

  const { onFail, datetype = 'text', onSuccess } = dataObject;

  xhr.onreadystatechange = function() {
    const { readyState, status } = xhr;
    if (readyState === 4) {
      if (status === 200) {
        if (datetype === 'text') {
          onSuccess && onSuccess(xhr.responseText);
        }
        if (datetype === 'xml') {
          onSuccess && onSuccess(xhr.responseXML);
        }
        if (datetype === 'json') {
          onSuccess && onSuccess(JSON.parse(xhr.responseText));
        }
      } else {
        onFail && onFail(xhr.responseText);
      }
    }
  };
  console.log('------', xhr);
  return xhr;
}

// export default request;
