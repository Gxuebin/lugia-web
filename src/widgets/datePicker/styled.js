import styled from 'styled-components';
import colorsFunc from '../css/stateColor';
import { px2emcss } from '../css/units';
const em = px2emcss(1.2);
const distance = {
  iconLeft: 10,
};
const { hoverColor } = colorsFunc();
export const Date = styled.span`
  display: inline-block;
  position: relative;
`;
export const DateInput = styled.input`
  padding: 0;
  margin: 0;
  width: 198px;
  height: 28px;
  border: 1px solid #ddd;
  padding-left: ${em(23)};
`;
export const Icons = styled.span`
  position: absolute;
  left: ${em(distance.iconLeft)};
  top: 50%;
  transform: translateY(-50%);
`;
export const DateWrapper = styled.div`
  width: ${props => props.width}px;
  padding: 30px 30px 44px;
  border: 1px solid #ddd;
  position: absolute;
  left: 0;
  top: 32px;
`;
export const DateWInner = styled.div`
  width: ${props => props.width}px;
`;
export const DateHeader = styled.div`
  font-size: 12px;
`;
export const HeaderTop = styled.div`
  text-align: center;
  margin-bottom: 24px;
`;
export const HeaderTopText = styled.span`
  color: #333;
  font-size: 14px;
  font-weight: 500;
  cursor: pointer;
  &:hover {
    color: ${hoverColor};
  }
`;
export const HeaderTopArrow = HeaderTopText.extend`    
    float:${props => props.position};
    margin-${props => props.position}:${props => props.margin}px;
    vertical-align:middle;
`;
export const HeaderWeekBox = styled.ul`
  padding: 0;
  margin: 0;
  margin-bottom: 12px;
`;
export const HeaderWeek = styled.li`
  border: 1px solid transparent;
  display: inline-block;
  width: ${props => (props.width - 14) / 7}px;
  height: ${props => (props.width - 14) / 7}px;
  text-align: center;
  line-height: ${props => (props.width - 14) / 7}px;
`;
export const DatePanel = styled.div`
  font-size: 12px;
`;
const dateSize = {
  DateChildWidth: 26,
};
export const DateChild = styled.span`
  border: 1px solid ${props => (props.isToday ? '#684fff' : 'transparent')};
  ${props => (props.isToday ? 'border-radius:50%;' : '')};
  display: inline-block;
  width: ${em(dateSize.DateChildWidth)};
  height: ${em(dateSize.DateChildWidth)};
  text-align: center;
  line-height: ${em(dateSize.DateChildWidth)};
  vertical-align: middle;
  margin-right: ${props => (props.width - (dateSize.DateChildWidth + 2) * 7) / 14}px;
  margin-left: ${props => (props.width - (dateSize.DateChildWidth + 2) * 7) / 14}px;
  &:nth-child(7n) {
    margin-right: 0;
  }

  cursor: pointer;
  &:hover {
    background: ${hoverColor};
  }

  color: ${props => (props.outMonth ? '#ccc' : '#666')};
  ${props => getDateChildStyle(props).chooseStyle};
`;
const getDateChildStyle = props => {
  const { choseDayIndex } = props;
  const chooseStyle = `  
        &:nth-child(${choseDayIndex}){
            background:#684fff;
            color:#fff;
            border-radius:50%;
        }  
        `;
  return {
    chooseStyle,
  };
};
