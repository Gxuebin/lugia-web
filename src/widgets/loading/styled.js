import styled, { keyframes } from 'styled-components';
import { px2emcss } from '../css/units';
import { FontSize, FontSizeNumber } from '../css/index';

const em = px2emcss(FontSizeNumber);
type CssProps = {
  circleDiameter?: number,
  width?: number,
  color?: string,
  delay?: number,
  scale?: boolean,
};
export const WrapperKeyframes = keyframes`
  0% {
    transform: rotate(0deg);
  }
  80% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;
export const circleKeyframes = keyframes`
  0% {
    transform: scale (1, 1);
    opacity: 0.5;
  }
  40% {
    transform: scale (1, 1);
    opacity: 1;
  }
  60% {
    transform: scale (1, 1);
    opacity: 1;
  }
  100% {
    transform: scale(1.2);
    opacity: 0.5;
  }
`;
const getStyled = (props?: CssProps) => {
  const { width, color, circleDiameter, delay, scale } = props;
  const LodingWrapperStyle = `
      font-size:${FontSize};
      position: absolute;
      width:${em(width)};
      height:${em(width)};
      left:50%;
      top:50%;
      transform: translate(-50%,-50%);
      -webkit-transform:translate(-50%,-50%);
  `;
  const LodingInnerStyle = `
      height: ${em(width)};
      width: ${em(circleDiameter)};
      position: absolute;
      left: ${em(width / 2 - circleDiameter / 2)};
      top: 0;
      transform:translateY(-50%);
      -webkit-transform:translateY(-50%);
      transform: rotate(0deg);
      animation: ${WrapperKeyframes} 2.5s infinite;
      animation-timing-function: cubic-bezier(0.28, 0.8, 0.69, 1);
      animation-delay: ${delay}s;
  `;
  const LodingInnerCircleStyle = `
    display: block;
    width: ${em(circleDiameter)};
    height: ${em(circleDiameter)};
    background: ${color};
    border-radius: 50%;
    ${
      scale
        ? `transform:scale(1.2);
          animation:${circleKeyframes} 2.5s infinite;
          animation-delay: 0.6s;
        `
        : `
        transform:scale(1);
        `
    }
  `;
  return {
    LodingWrapperStyle,
    LodingInnerStyle,
    LodingInnerCircleStyle,
  };
};
export const LodingWrapper = styled.div`
  ${props => getStyled(props).LodingWrapperStyle};
`;
export const LodingInner = styled.div`
  ${props => getStyled(props).LodingInnerStyle};
`;
export const LodingInnerCircle = styled.span`
  ${props => getStyled(props).LodingInnerCircleStyle};
`;
