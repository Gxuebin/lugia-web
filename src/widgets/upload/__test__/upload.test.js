/*
 *@flow
 *
 */

import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Upload, { getIndexInArray, isKeyInArray, getPercentValue } from '../upload';
import { getRequestXHR, getStringFromObject, getParamsData } from '../request';
import { getIconByType, getListIconType } from '../getelement';
import Enzyme, { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Widget from '../../consts/index';
import 'jest-styled-components';
import Icon from '../../icon';
import styled from 'styled-components';
const { mockObject, VerifyOrder, VerifyOrderConfig } = require('@lugia/jverify');

Enzyme.configure({ adapter: new Adapter() });

describe('Rate Test', () => {
  const target = mount(<Upload url={'xxxx.test'} />);
  it('css', () => {
    const target = <Upload url={'xxxx.test'} />;
    expect(renderer.create(target).toJSON()).toMatchSnapshot();
  });

  it('props autoUpload true', () => {
    const target = mount(<Upload autoUpload={true} url={'xxxx.test'} />);
    expect(target.state().isAllowUpload).toEqual(true);
  });

  it('props autoUpload false', () => {
    const target = mount(<Upload autoUpload={false} url={'xxxx.test'} />);
    expect(target.state().isAllowUpload).toEqual(false);
  });

  it('props url', () => {
    const target = mount(<Upload url={'xxxx.test'} />);
    expect(target.props().url).toEqual('xxxx.test');
  });

  it('props fileList', () => {
    const target = mount(
      <Upload
        url={'xxxx.test'}
        fileList={[
          { id: 1, name: '文件11111.jpg', status: 'done' },
          { id: 2, name: '文件666.doc', status: 'fail' },
        ]}
      />
    );
    expect(target.state().fileListDone).toEqual([
      { id: 1, name: '文件11111.jpg', status: 'done' },
      { id: 2, name: '文件666.doc', status: 'fail' },
    ]);
  });

  it('props fileList null', () => {
    const target = mount(<Upload url={'xxxx.test'} />);
    expect(target.state().fileListDone).toEqual([]);
  });

  function checkFindIndex(data: Array<string>, key: string, expectation: number) {
    it('Function getIndexInArray ', () => {
      const res = getIndexInArray(data, key.toLowerCase());
      expect(res).toEqual(expectation);
    });
  }
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jpg', 0);
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jPg', 0);
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'JPG', 0);
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jpEg', 2);
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'ABC', -1);
  checkFindIndex(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'BMP', 5);

  function checkKeyInArr(data: Array<string>, key: string, expectation: boolean) {
    it('Function isKeyInArr ', () => {
      const res = isKeyInArray(data, key.toLowerCase());
      expect(res).toEqual(expectation);
    });
  }
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jpg', true);
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jPg', true);
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'JPG', true);
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'jpEg', true);
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'ABC', false);
  checkKeyInArr(['jpg', 'png', 'jpeg', 'gif', 'svg', 'bmp'], 'BMP', true);

  function checkGetPercentValue(current: ?number, total: ?number, expectation: number) {
    it('Function getPercentValue ', () => {
      const res = getPercentValue(current, total);
      expect(res).toEqual(expectation);
    });
  }
  checkGetPercentValue(2345, 3455, 67);
  checkGetPercentValue(null, 3455, 0);
  checkGetPercentValue(2345, undefined, 0);
  checkGetPercentValue(3488, 3455, 100);

  function checkGetListIconType(fileName: ?string, expectation: string) {
    it('Function getListIconType ', () => {
      const res = getListIconType(fileName);
      expect(res).toEqual(expectation);
    });
  }
  checkGetListIconType('http://jshdfj.jpg', 'picture');
  checkGetListIconType('http://jshdfj.png', 'picture');
  checkGetListIconType('http://jshdfj.JPEG', 'picture');
  checkGetListIconType('http://jshdfj.avi', 'video');
  checkGetListIconType('http://jshdfj.MP4', 'video');
  checkGetListIconType('http://jshdfj.cccc', 'file');
  checkGetListIconType('http://jshdfj.text', 'file');
  checkGetListIconType('http://jshdfj.doc', 'file');
  checkGetListIconType('', 'file');
  checkGetListIconType(null, 'file');
  checkGetListIconType(undefined, 'file');

  it('Function getRequestXHR ', () => {
    const res = getRequestXHR();
    expect(res instanceof window.XMLHttpRequest).toEqual(true);
  });

  function checkGetStringFromObject(data: ?Object, expectation: string) {
    it('Function getStringFromObject ', () => {
      const res = getStringFromObject(data);
      expect(res).toEqual(expectation);
    });
  }
  checkGetStringFromObject({ u: 23, name: 'lll', file: 'xsdss' }, 'u=23&name=lll&file=xsdss');
  checkGetStringFromObject({ cc: 66, df: 'lll', file: 'xsdss' }, 'cc=66&df=lll&file=xsdss');
  checkGetStringFromObject(null, '');
  checkGetStringFromObject(undefined, '');

  function checkGetParamsData(data: Object) {
    it('Function getParamsData ', () => {
      const res = getParamsData(data, {});
      expect(res instanceof FormData).toEqual(true);
    });
  }
  checkGetParamsData({ data: { a: 123, b: 223 }, name: 'file', file: '666' });
  checkGetParamsData({ data: { a: 777, b: 888 }, name: 'abc', file: '567' });

  function checkGetIconByType(status: ?string, expectation: ?string, props?: Object = {}) {
    it('Function GetIconByType ', () => {
      const res = getIconByType(status, props);
      const { type } = props;
      if (!status) {
        expect(res).toBe(expectation);
      } else if (type === 1 && status === 'default') {
        expect(res).toEqual(expectation);
      } else {
        const { iconClass } = res.props;
        expect(iconClass).toBe(expectation);
      }
    });
  }
  checkGetIconByType('default', '上传', { type: 1 });
  checkGetIconByType('default', 'lugia-icon-financial_upload right');
  checkGetIconByType('loading', 'lugia-icon-financial_loading_o loadIcon');
  checkGetIconByType(undefined, null);
  checkGetIconByType(undefined, null, { type: 1 });

  function setStateValue(props: Object, expectation: Object) {
    it('Function setStateValue ', () => {
      target.instance().setStateValue(props);
      expect(target.state()).toEqual(expectation);
    });
  }
  setStateValue(
    { classNameStatus: 'done', defaultText: '文件已上传成功！' },
    {
      classNameStatus: 'done',
      defaultText: '文件已上传成功！',
      fileListDone: [],
      isAllowUpload: true,
    }
  );
  setStateValue(
    { classNameStatus: 'loading', defaultText: '文件正在上传！' },
    {
      classNameStatus: 'loading',
      defaultText: '文件正在上传！',
      fileListDone: [],
      isAllowUpload: true,
    }
  );
  setStateValue(
    { classNameStatus: 'default', defaultText: '文件已上传成功！', abc: 'abc' },
    {
      abc: 'abc',
      classNameStatus: 'default',
      defaultText: '文件已上传成功！',
      fileListDone: [],
      isAllowUpload: true,
    }
  );

  function checkGetFileList(
    props: Object | number,
    expectation: Array<Object>,
    data?: Array<Object>
  ) {
    it('Function getFileList ', () => {
      const res = target.instance().getFileList(props, data);
      expect(res).toEqual(expectation);
    });
  }
  checkGetFileList({ id: 1, name: '文件11111.jpg', status: 'loading' }, [
    { id: 1, name: '文件11111.jpg', status: 'loading' },
  ]);
  checkGetFileList({ id: 2, name: '文件22222222.jpg', status: 'loading' }, [
    { id: 1, name: '文件11111.jpg', status: 'loading' },
    { id: 2, name: '文件22222222.jpg', status: 'loading' },
  ]);
  checkGetFileList(
    1,
    [
      { id: 1, name: '文件11111.jpg', status: 'loading', percent: 20 },
      { id: 2, name: '文件22222222.jpg', status: 'loading' },
    ],
    [{ target: 'percent', value: 20 }]
  );
  checkGetFileList(
    3,
    [
      { id: 1, name: '文件11111.jpg', status: 'loading', percent: 20 },
      { id: 2, name: '文件22222222.jpg', status: 'loading' },
    ],
    [{ target: 'percent', value: 50 }]
  );

  function uploadProgress(props: Object, expectation: string, expectation2: Array<Object>) {
    it('Function uploadProgress ', () => {
      target
        .instance()
        .setStateValue({ fileListDone: [{ id: 1, name: '文件11111.jpg', status: 'default' }] });
      target.instance().uploadProgress(props, 1);
      expect(target.state().classNameStatus).toEqual(expectation);
      expect(target.state().fileListDone).toEqual(expectation2);
    });
  }
  uploadProgress({ loaded: 0, total: 2048 }, 'loading', [
    { id: 1, name: '文件11111.jpg', status: 'loading', percent: 0 },
  ]);
  uploadProgress({ loaded: 1024, total: 2048 }, 'loading', [
    { id: 1, name: '文件11111.jpg', status: 'loading', percent: 50 },
  ]);
  uploadProgress({ loaded: 2048, total: 2048 }, 'loading', [
    { id: 1, name: '文件11111.jpg', status: 'loading', percent: 100 },
  ]);
  uploadProgress({}, 'loading', [{ id: 1, name: '文件11111.jpg', status: 'loading', percent: 0 }]);
  function uploadSuccess(props: Object, expectation: string, expectation2: Array<Object>) {
    it('Function uploadSuccess ', () => {
      target
        .instance()
        .setStateValue({ fileListDone: [{ id: 1, name: '文件11111.jpg', status: 'default' }] });
      target.instance().uploadSuccess(props, 1);
      expect(target.state().classNameStatus).toEqual(expectation);
      expect(target.state().fileListDone).toEqual(expectation2);
    });
  }
  uploadSuccess({ currentTarget: {} }, 'done', [{ id: 1, name: '文件11111.jpg', status: 'done' }]);

  function isIdInArray(id: number, props: Array<Object>, expectation: boolean) {
    it('Function isIdInArray ', () => {
      const res = target.instance().isIdInArray(id, props);
      expect(res).toEqual(expectation);
    });
  }
  isIdInArray(1, [{ id: 1, name: '文件11111.jpg', status: 'default' }], true);
  isIdInArray(2, [{ id: 1, name: '文件11111.jpg', status: 'default' }], false);
  isIdInArray(1, [], false);

  function setAutoUploadState(
    props: boolean,
    choosedFileList: ?Array<Object>,
    expectation: boolean
  ) {
    it('Function setAutoUploadState ', () => {
      target.instance().setStateValue({ choosedFile: choosedFileList });
      target.instance().setAutoUploadState(props);
      expect(target.state().isAllowUpload).toEqual(expectation);
    });
  }
  setAutoUploadState(false, undefined, true);
  setAutoUploadState(true, [new FormData()], true);
  setAutoUploadState(false, [new FormData()], false);

  function setDeleteList(index: number, expectation: Array<Object>) {
    it('Function setDeleteList ', () => {
      target
        .instance()
        .setStateValue({
          fileListDone: [
            { id: 1, name: '文件11111.jpg', status: 'done' },
            { id: 2, name: '文件2222.jpg', status: 'default' },
          ],
        });
      target.instance().setDeleteList(index);
      expect(target.state().fileListDone).toEqual(expectation);
    });
  }
  setDeleteList(1, [{ id: 1, name: '文件11111.jpg', status: 'done' }]);
  setDeleteList(0, [{ id: 2, name: '文件2222.jpg', status: 'default' }]);
  setDeleteList(2, [
    { id: 1, name: '文件11111.jpg', status: 'done' },
    { id: 2, name: '文件2222.jpg', status: 'default' },
  ]);

  function checkUploadFail(
    props: Object,
    id: number,
    expectation: string,
    expectation2: Array<Object>
  ) {
    it('Function checkUploadFail ', () => {
      target
        .instance()
        .setStateValue({ fileListDone: [{ id: 1, name: '文件11111.jpg', status: 'loading' }] });
      target.instance().uploadFail(props, id);
      expect(target.state().classNameStatus).toEqual(expectation);
      expect(target.state().fileListDone).toEqual(expectation2);
    });
  }
  checkUploadFail({}, 1, 'fail', [{ id: 1, name: '文件11111.jpg', status: 'fail' }]);
  checkUploadFail({}, 2, 'fail', [{ id: 1, name: '文件11111.jpg', status: 'loading' }]);

  const files = [
    {
      lastModified: 1541066150245,
      lastModifiedDate: new Date(1541066150245),
      name: 'headPic.jpg',
      size: 53021,
      type: 'image/jpeg',
      webkitRelativePath: '',
    },
  ];
  function checkSetChoosedFile(expectation: string) {
    it('Function checkSetChoosedFile ', () => {
      target.instance().setChoosedFile(files);
      expect(target.state().choosedFile).toEqual(expectation);
      target.instance().getPreviewInfo(new Blob(files));
      expect(target.state().previewUrl).toEqual(expectation);
    });
  }
  checkSetChoosedFile(files);

  function checkGetPreviewInfo(expectation: string) {
    it('Function getPreviewInfo ', () => {
      target.instance().setChoosedFile(files);
      expect(target.state().choosedFile).toEqual(expectation);
    });
  }
});
