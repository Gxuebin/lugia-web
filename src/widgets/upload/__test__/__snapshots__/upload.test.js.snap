// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate Test css 1`] = `
.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c2 {
  width: 100%;
  display: none;
}

.c1 {
  width: 366px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

.c3 {
  width: 346px;
  height: 30px;
  border: 1px solid #e8e8e8;
  border-radius: 4px;
  color: #ccc;
  padding: 0 0 0 10px;
  line-height: 30px;
  overflow: hidden;
  box-sizing: border-box;
  position: relative;
}

.c3.done {
  border: 1px solid #684fff;
  color: #333;
  position: relative;
}

.c3.done i {
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  top: 50%;
  right: 10px;
}

.c3.loading {
  border: 1px solid #684fff;
}

.c3.loading .loadIcon {
  margin-right: 10px;
  color: #684fff;
  -webkit-animation: iVXCSc 0.8s linear infinite;
  animation: iVXCSc 0.8s linear infinite;
}

.c3.hasBtn {
  border-radius: 4px 0 0 4px;
  border: 1px solid #9482ff;
  width: 286px;
}

.c3 i.right {
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  top: 50%;
  right: 10px;
}

.c3 i.success {
  color: #56c22d;
}

.c3 i.error {
  color: #f22735;
}

.c4.loadIcon {
  margin-right: 10px;
}

.c0 {
  width: 366px;
  position: relative;
  padding: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <input
      accept="*"
      className="c2"
      id="upload"
      multiple={false}
      onChange={[Function]}
      type="file"
    />
    <div
      className="default c3"
      onClick={[Function]}
    >
      <i
        className="lugia-icon-financial_upload right c4 c5"
      />
       
      请将文件拖到此处
    </div>
  </div>
   
</div>
`;
